src/Framework/Framework/Controls/ListBox.cs

```diff
@@ -32,16 +60,54 @@ protected override void AddAttributesToRender(IHtmlWriter writer, IDotvvmRequest
             writer.AddKnockoutDataBind("size", this, SizeProperty, () => writer.AddAttribute("size", Size.ToString()));
         }
 
-        /// <summary>
-        /// Gets or sets number of rows visible in this ListBox.
-        /// </summary>
-        public int Size
+        protected override void RenderSelectedValueProperty(IHtmlWriter writer)
         {
-            get { return (int)GetValue(SizeProperty)!; }
-            set { SetValue(SizeProperty, value); }
+            if (SelectionMode == ListBoxSelectionMode.Multiple)
+            {
+                writer.AddAttribute("multiple", "multiple");
+                writer.AddKnockoutDataBind("selectedOptions", this, SelectedValueProperty, renderEvenInServerRenderingMode: true);
+            }
+            else
+            {
+                writer.AddKnockoutDataBind("value", this, SelectedValueProperty, renderEvenInServerRenderingMode: true);
+            }
+            writer.AddKnockoutDataBind("valueAllowUnset", "true");
         }
 
-        public static readonly DotvvmProperty SizeProperty =
-            DotvvmProperty.Register<int, ListBox>(t => t.Size, defaultValue: 10);
+        [ControlUsageValidator(Override = true)]
+        public static new IEnumerable<ControlUsageError> ValidateUsage(ResolvedControl control)
```

Comment:
Can you please add some tests for the usage validation? 
I am not sure what will happen if the SelectionMode=Multiple and the SelectedValues is not IEnumerable - the GetEnumerableType will probably throw an exception so this method will fail to.

Reply:
good catch you are right

Reply:
Done ;)